src/Converter/Feature2Texture.js

```diff
@@ -40,15 +42,19 @@ function _drawPolygon(ctx, vertices, indices, style, size, extent, invCtxScale,
     }
 
     // fill polygon only
-    if (canBeFilled && style.fill.color) {
-        fillStyle(style, ctx);
+    if (canBeFilled && (style.fill.color || style.fill.pattern)) {
+        fillStyle(style, ctx, invCtxScale);
         ctx.fill();
     }
 }
 
-function fillStyle(style, ctx) {
-    if (ctx.fillStyle !== style.fill.color) {
+function fillStyle(style, ctx, invCtxScale) {
+    if (style.fill.color && ctx.fillStyle !== style.fill.color) {
         ctx.fillStyle = style.fill.color;
+    } else if (style.fill.pattern && ctx.fillStyle.src !== style.fill.pattern.src) {
```

Comment:
Shouldn't this be the first option to check ?

Reply:
yes